arelle/plugin/loadFromOIM.py

```diff
@@ -2106,6 +2108,12 @@ def checkDim(tblTmpl, dimName, dimValue, *pathSegs):
         
         numFactCreationXbrlErrors = 0
         
+        taxonomyContextRefs = {
+            modelRelationship.contextElement
+            for hasHypercubeRelationship in (XbrlConst.all, XbrlConst.notAll)
```

Comment:
This seems kind of unfortunately slow in some cases. It's probably a lot more difficult of a fix but it seems like it'd maybe be more cost effective to do in the DTS parsing code itself. However, this does fix the bug so I won't push very hard on this. It'd be nice to see a test if possible